src/Event/EventDispatcher/SyncEventDispatcher.php

```diff
@@ -37,38 +37,33 @@
 
 use function get_class;
 use Infection\Event\Subscriber\EventSubscriber;
+use function method_exists;
+use function Safe\substr;
+use function strrchr;
 
 /**
  * @internal
  */
 final class SyncEventDispatcher implements EventDispatcher
 {
     /**
-     * @var callable[][]
+     * @var EventSubscriber[]
      */
-    private $listeners = [];
+    private $subscriber = [];
 
     public function dispatch(object $event): void
     {
-        $name = get_class($event);
+        $subscription = 'on' . substr((string) strrchr(get_class($event), '\\'), 1);
```

Comment:
Could event itself know this method name? And memoize it.

Reply:
That could be doable but I think it would be better done in an abstract class `Event` which all events should extend then